### PR TITLE
Mac fixes

### PR DESCRIPTION
**Pull request by jherico**
## Note

This PR can be QA'd and dev approved but should not be merged until ops enables Qt 5.9.1 builds for mac for the main development build

## Fixes

* Rendering freezing when executing a handshake
* A crash on startup
* A potential crash on leaving a domain with a text entity
* A potential crash on shutdown (there may be others)

## Testing

With a mac client and a fresh install, start up the application.  On clean installs with the current master this may always crash.  

Uses the X command on the keyboard to extend your hand for a handshake and shake hands with another avatar who is doing the same.  On master builds this will cause the rendering to simply stop rendering, while the application continues running.  On this build it should not freeze.

This build may still crash on shutdown, but I want to defer that to another PR if it does.  It may also have been something specific to a debug build that triggered that crash.

### CODE REVIEW

File: interface/src/main.cpp [hunk 1]
```diff
@@ -144,25 +144,33 @@ int main(int argc, const char* argv[]) {
...
 
+
+    // FIXME this method of checking the OpenGL version screws up the `QOpenGLContext::globalShareContext()` value, which in turn
+    // leads to crashes when creating the real OpenGL instance.  Disabling for now until we come up with a better way of checking
+    // the GL version on the system without resorting to creating a full Qt application
+#if 0
     // Check OpenGL version.
     // This is done separately from the main Application so that start-up and shut-down logic within the main Application is
     // not made more complicated than it already is.
-    bool override = false;
```
Comment (jherico) [inline interface/src/main.cpp]: override is a C++ keyword.  Don't use as variable name

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3131/HighFidelity-Beta-PR11323-eb8b892571df62edcdfe44ff4e8c8a31ca9a5ef5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3131/HighFidelity-Beta-PR11323-eb8b892571df62edcdfe44ff4e8c8a31ca9a5ef5.dmg

**Comment by conklin94122** [comment]
Still crashes on start. Fail.

**Comment by jherico** [comment]
@conklin94122 looks like some Qt 5.9 dependencies slipped into the QML, which means that the mac builds are failing because the build machines are still using Qt 5.6 for mac.  I've asked ops to switch to 5.9 for the Mac builds.  as soon as this PR passes we can migrate the main builds and then merge this.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3160/HighFidelity-Beta-PR11323-eb8b892571df62edcdfe44ff4e8c8a31ca9a5ef5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3160/HighFidelity-Beta-PR11323-eb8b892571df62edcdfe44ff4e8c8a31ca9a5ef5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3162/HighFidelity-Beta-PR11323-48b6586d01f5f4eb27af3ad04b6ff80d2ab43e8d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3162/HighFidelity-Beta-PR11323-48b6586d01f5f4eb27af3ad04b6ff80d2ab43e8d.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3167/HighFidelity-Beta-PR11323-48b6586d01f5f4eb27af3ad04b6ff80d2ab43e8d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3167/HighFidelity-Beta-PR11323-48b6586d01f5f4eb27af3ad04b6ff80d2ab43e8d.dmg

**Comment by conklin94122** [comment]
@jherico Is this still waiting for a code review before being merged? (cc: @cozza13)

**Pull request merged**